src/Surfnet/StepupRa/RaBundle/VettingProcedure.php

```diff
@@ -239,7 +258,7 @@ private function isReadyForSecondFactorToBeVerified()
      */
     private function isReadyForIdentityVerification()
     {
-        return $this->inputSecondFactorIdentifier === $this->secondFactor->secondFactorIdentifier
+        return ($this->inputSecondFactorIdentifier === $this->secondFactor->secondFactorIdentifier || $this->skipProvePossession)
```

Comment:
The addition of the optional condition makes the already hard to understand expression even worse. 

The main confusion lies in the first part of the expression. For me `$this->inputSecondFactorIdentifier === $this->secondFactor->secondFactorIdentifier` is imediately clears this is a token that is ready for verification.

Maybe this condition can be made more clear using a couple of comments. Alternatively it could be wrapped in a more expressive private method.

This also goes for line 270

Reply:
Good suggestion